### PR TITLE
Fill openvsx-sync plugins list from che-theia-plugins.yaml

### PR DESCRIPTION
**Pull request by vinokurig**
### What does this PR do?
Fill openvsx-sync plugins list from che-theia-plugins.yaml

List of extensions that are present in the `che-theia-plugins.yaml` but are not published in the https://open-vsx.org:
https://github.com/eclipse-cdt/cdt-vscode
https://github.com/redhat-developer/netcoredbg-theia-plugin
https://github.com/redhat-developer/omnisharp-theia-plugin
https://github.com/pizzafactory-contorno/vscode-libra-move
https://github.com/MicroShed/mp-starter-vscode-ext
https://github.com/SebastianZaha/vscode-emacs-friendly
https://github.com/sgwozdz/jenkinsfile-support

### What issues does this PR fix or reference?

fixes https://github.com/eclipse/che/issues/21621

### How to test this PR?

Run `./build.sh` script

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [ ] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [ ] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Comment by svor** [comment]
https://github.com/redhat-developer/netcoredbg-theia-plugin
and
https://github.com/redhat-developer/omnisharp-theia-plugin
are Theia plugins they could be replaced by https://open-vsx.org/extension/muhammad-sammy/csharp which is already in the list. I think we can skip them

**Comment by svor** [comment]
About others extension I don't see that any of our devfiles use them, maybe we can sip them as well
or ask owners to publish the extensions on OpenVSX

**Comment by svor** [comment]
Looking at the [PR check build](https://github.com/eclipse-che/che-plugin-registry/runs/7951484078?check_suite_focus=true), I see some errors during the publish process of vscode-openshift-connector, but the extension was published at the end :  
```
Downloading https://open-vsx.org/api/redhat/vscode-openshift-connector/0.4.0/file/redhat.vscode-openshift-connector-0.4.0.vsix into redhat folder...
2022-08-22 12:26:24.528  WARN 35 --- [io-9000-exec-10] org.eclipse.openvsx.ExtensionProcessor   : Failed to read entry

org.eclipse.openvsx.util.ErrorResultException: The file extension/out/tools/darwin/oc exceeds the size limit of 32 MB.
```
@benoitf do you know if it's something critical?

Also some comparisons: 

- The size of the image was increased by 3GB (now it's 4.3 GB)
- The build time on my local machine from the scratch was 28 minutes, on Jenkins - it's about 9 minutes

**Comment by benoitf** [comment]
@svor: what is the size of all .vsix files that we're adding ?

File: openvsx-sync.json [hunk 1]
```diff
@@ -1,6 +1,66 @@
...
+    "redhat.vscode-camelk",
     "redhat.java",
-    "redhat.vscode-xml"
+    "vscjava.vscode-java-debug",
+    "vscjava.vscode-java-test",
+    "redhat.vscode-microprofile",
+    "redhat.vscode-quarkus",
+    "redhat.fabric8-analytics",
+    "redhat.vscode-tekton-pipelines",
+    "redhat.project-initializer",
```
Comment (svor) [inline openvsx-sync.json:30]:
```suggestion
    "redhat.project-initializer",
    "redhat.vscode-redhat-account",
```
**redhat.vscode-openshift-connector** depends on **redhat.vscode-redhat-account**

**Comment by svor** [comment]
> @svor: what is the size of all .vsix files that we're adding ?

@benoitf  it's 710M

**Review by benoitf — APPROVED** [review]

**Comment by svor** [comment]
no image's size is 3.91GB

**Comment by benoitf** [comment]
@svor and compressed size ?

**Review by svor — APPROVED** [review]

**Comment by svor** [comment]
> @svor and compressed size ?

@benoitf sorry, haven't seen your message before, compressed one is ~ 2.2 Gb

**Pull request merged**